### PR TITLE
Bump the Node.js version and add .gitignore

### LINKED ISSUE #5: Node.js 16 is deprecated
### Summary

Node.js 16 is deprecated.

### Content

This action uses Node.js 16, which is now deprecated. In [skills/secure-code-game](/skills/secure-code-game), the workflow that uses the action raises warning.
Is this deprecated version usage is deliberate or should the version be updated?

### Discussed in https://github.com/skills/secure-code-game/discussions/87#discussioncomment-9048923

<div type='discussions-op-text'>

<sup>Originally posted by **sinsukehlab** April  9, 2024</sup>
Yes, I refer to Season 2 Level 1.
The action [dduzgun-security/secure-code-game-action](/dduzgun-security/secure-code-game-action) used in the workflow jarvis-code.yml still uses Node.js 16.
https://github.com/dduzgun-security/secure-code-game-action/blob/b30b00ac91e4cbd0a8636ccbef1f38f4ffc376ea/action.yml#L16-L18
```yml
runs:
  using: 'node16'
  main: 'index.js'
```
</div>

### PR DESCRIPTION
**Pull request by sinsukehlab**
### Summary

Bump the Node.js version from 16 to 20 and add .gitignore

### Changes

- node16 &rarr; node20
- Create .gitignore (Node template)

Closes: #5

### Task list

- [x] For workflow changes, I have verified the Actions workflows function as expected.
- [x] For content changes, I have reviewed the [style guide](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by jkcso** [comment]
@dduzgun-security feel free to merge if you approve :-)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+bower_components
+
+# node-waf configuration
+.lock-wscript
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
```
Comment (dduzgun-security) [inline .gitignore]: I think we need to keep the `node_modules/` looking here: https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github

```suggestion
# node_modules/
```

**Review by dduzgun-security — CHANGES_REQUESTED** [review]

**Review by jkcso — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (sinsukehlab) [reply .gitignore]: Got it. This repo currently needs `node_modules/` directory.
https://github.com/sinsukehlab/workflow-test/actions/runs/8717720558/job/23913490711

**Comment by sinsukehlab** [comment]
@dduzgun-security Now you can merge this PR.
I will open another PR to introduce [vercel/ncc](/vercel/ncc).

**Review by dduzgun-security — APPROVED** [review]

**Comment by sinsukehlab** [comment]
I believe that this commit should be tagged with `v1.1.0`.

**Pull request merged**